### ISSUE TITLE
Mentions plugin: Deleting part of mention (backspace) throws error - worse for async. resolver

### ISSUE THREAD

**Issue by FlowIT-JIT** [ckeditor/ckeditor4#5036]
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1) Open https://jsfiddle.net/n2kLv40c/2/ in a new tab
2) On a new line in the editor, write: @James
3) _Very_ quickly press backspace 3 times
4) The result is shown in the screenshot below - CKEditor throws an error

![image](https://user-images.githubusercontent.com/10479064/149519134-6746d5fa-26b6-419b-b81c-107d0b06d1c0.png)

The problem gets even worse when resolving mentions async.

1) Open https://jsfiddle.net/L8nmrq49/ in a new tab
2) On a new line in the editor, write: @James
3) _Very_ quickly press backspace 3 times
4) The result is shown in the screenshot below - CKEditor now throws two errors.
The first error is thrown when backspace has been pressed twice, the second error follows when backspace is pressed yet again.

![image](https://user-images.githubusercontent.com/10479064/149519534-9141bb69-ac52-4042-a3f4-4f316768ab52.png)

## Other details

* Browser: Chrome Version 96.0.4664.110 (Official Build) (x86_64)
* OS: macOS 10.13.6 (17G14042)
* CKEditor version: 4.17.1 (Full-all)
* Installed CKEditor plugins: …

**Comment by FlowIT-JIT** [comment]
I posted another bug related to the mentions plugin three days ago. Perhaps worth looking into at the same time.
https://github.com/ckeditor/ckeditor4/issues/5028

**Comment by KarolDawidziuk** [comment]
Hi, @FlowIT-JIT 
Thank you for the report.
 
I can confirm this issue.

A first look at the code shows that this could be a throttling problem.
https://github.com/ckeditor/ckeditor4/blob/master/plugins/mentions/plugin.js#L117

When we change the value from `200` to a lower(e.g. `50`) - the problem disappears.

**Comment by FlowIT-JIT** [comment]
@KarolDawidziuk Thanks for the work around 👍😊

**Comment by jacekbogdanski** [comment]
@KarolDawidziuk could you check if this issue has not been reported already? I have some feelings that we already confirmed it some time ago.

**Comment by KarolDawidziuk** [comment]
Yes, there was such a report, but this one has more details about the issue so I closed the old one. 
So this issue is no longer blocked.
